presto-main/src/main/java/io/prestosql/execution/SqlQueryManager.java

```diff
@@ -316,4 +326,21 @@ private void enforceCpuLimits()
             }
         }
     }
+
+    /**
+     * Enforce max raw data size at the query level
+     */
+    public void enforceQueryMaxDataSizeLimits()
+    {
+        queryTracker.getAllQueries().stream()
```

Comment:
It would be better to use min value of QueryManagerConfig and SystemSessionProperties to avoid overridden by session as maxQueryCpuTime.

If the property isn't configured, we don't need to iterate queries. 

Also, this method chain looks too nested. Could you use for-each loop?

Reply:
[For Another Review] Do you think we could apply this logic of taking the min to avoid session override of sever limits to time limits in enforceTimeLimits() too ?